migrations/helpers/dart_samples_update_helper.py

```diff
@@ -0,0 +1,387 @@
+                "$match": {
+                    FIELD_ROOT_SAMPLE_ID: {"$nin": cp_samples_df[FIELD_ROOT_SAMPLE_ID].to_list()},  # type: ignore
+                }
+            }
+
+            pipeline.append(second_match)  # type: ignore
+
+            # get the samples between those dates minus the cherry-picked ones
+            results = samples_collection.aggregate(pipeline)
+            samples = list(results)
+
+            print(
+                f"{len(samples)} documents found in the mongo database between these timestamps and not cherry-picked"
+            )
+
+            # 3. add the relevant filtered positive fields if not present
+            filtered_positive_identifier = FilteredPositiveIdentifier()
+            version = filtered_positive_identifier.current_version()
+            update_timestamp = datetime.now()
+
+            print("Updating filtered positives...")
+
+            update_filtered_positive_fields(
+                filtered_positive_identifier,
+                samples,
+                version,
+                update_timestamp,
+            )
+
+            print("Updated filtered positives")
+
+            # 4. add the UUID fields if not present
+            add_sample_uuid_field(samples)
+
+            samples_updated_with_source_plate_uuids(mongo_db, samples)
+
+            # 5. update samples in mongo updated in either of the above two steps (would expect the same set of samples
+            #       from both steps)
+            print("Updating Mongo...")
+            _ = update_mongo_fields(mongo_db, samples, version, update_timestamp)
+            print("Finished updating Mongo")
+
+        # 6. update the MLWH (should be an idempotent operation)
+        # convert mongo field values into MySQL format
+        for sample in samples:
+            mongo_docs_for_sql.append(map_mongo_doc_to_sql_columns(sample))
+
+        if number_docs_found > 0:
+            print(f"Updating MLWH database for {len(mongo_docs_for_sql)} sample documents")
+            # create connection to the MLWH database
+            with create_mysql_connection(config, False) as mlwh_conn:
+
+                # execute sql query to insert/update timestamps into MLWH
+                run_mysql_executemany_query(mlwh_conn, SQL_MLWH_MULTIPLE_INSERT, mongo_docs_for_sql)
+        else:
+            print("No documents found for this timestamp range, nothing to insert or update in MLWH")
+
+        # 7. add all plates to DART, but only RESULT=Positive samples
+        update_dart_filtered_positive_fields(config, samples)
+    except Exception as e:
+        print(e)
+
+
+def add_sample_uuid_field(samples: List[Sample]) -> List[Sample]:
+    # add lh sample uuid
+    for sample in samples:
+        if FIELD_LH_SAMPLE_UUID not in [*sample]:
+            sample[FIELD_LH_SAMPLE_UUID] = str(uuid.uuid4())
+
+    return samples
+
+
+def update_mongo_fields(mongo_db, samples: List[Sample], version: str, update_timestamp: datetime) -> bool:
```

Comment:
Looks like you've borrowed some logic from the filtered positive migration. Worth extracting out to a shared location?

Reply:
As per your previous comment, resolving this on this PR, but still a valid option to extract out shared methods used both here and in the update filtered positives migration